textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/reader/DataReader.java

```diff
@@ -74,18 +107,77 @@ public ITuple getNextTuple() throws DataFlowException {
             if(cursor >= scoreDocs.length){
                 return null;
             }
-            Document document = luceneIndexSearcher.doc(scoreDocs[cursor++].doc);
-            
+            Document document = luceneIndexSearcher.doc(scoreDocs[cursor].doc);
+            List<Span> spanList = new ArrayList<>();
             List<IField> fields = new ArrayList<IField>();
-            Schema schema = dataReaderPredicate.getDataStore().getSchema();
+
             for (Attribute  attr : schema.getAttributes()) {
                 FieldType fieldType = attr.getFieldType();
                 String fieldValue = document.get(attr.getFieldName());
                 fields.add(Utils.getField(fieldType, fieldValue));
             }
-            
-            DataTuple dataTuple = new DataTuple(schema, fields.toArray(new IField[fields.size()]));
+
+            // If the span Information is not requested,
+            // just return the dataTuple without span information.
+
+            if(!dataReaderPredicate.getIsSpanInformationAdded()){
+                cursor++;
+                DataTuple dataTuple = new DataTuple(schema, fields.toArray(new IField[fields.size()]));
+                return  dataTuple;
+            }
+
+            // Create span information.
+
+            for(Attribute attr: attributeList){
+
+                String fieldName  = attr.getFieldName();
+                // Get the term vector fot the current field.
+                Terms vector = luceneIndexReader.getTermVector(scoreDocs[cursor].doc,fieldName);
+
+                if (vector != null) {
+                    TermsEnum vectorEnum = vector.iterator();
+                    int queryTokenIndex = 0;
+                    // Search for all the query tokens in the term vector one by one.
+                    for(BytesRef term: queryTokensInBytesRef){
+
+                        //If Term is found, calculate the position info and add to the Spans
+                        if(vectorEnum.seekExact(term)){
+                            PostingsEnum postings = vectorEnum.postings(null, PostingsEnum.POSITIONS);
+
+                            while (postings.nextDoc() != DocIdSetIterator.NO_MORE_DOCS) {
+                                int freq = postings.freq();
+                                // Create a new span for every occurrence.
+                                while (freq-- > 0) {
+                                    int tokenOffset = postings.nextPosition();
+                                    int start = postings.startOffset();
+                                    int end = start+term.length;

```

Comment:
Per our discussion, should we do `int end = start+term.length + 1`?  @sandeepreddy602 @zuozhi